utils/adapters/firebase.ts

```diff
@@ -0,0 +1,65 @@
+import { App as FirebaseApp } from "firebase-admin/app";
+import { getAuth } from "firebase-admin/auth";
+import { getFirestore } from "firebase-admin/firestore";
+import { randomBytes, secretbox } from "tweetnacl";
+import SolanaAuth, { Adapter } from "../SolanaAuth";
+
+/** FROM: https://github.com/dchest/tweetnacl-js/wiki/Examples */
+const newNonce = () => randomBytes(secretbox.nonceLength);
+
+/**
+ * A SolanaAuth Adapter to connect with Firebase using Firestore
+ * @param {import("firebase-admin/app").App} firebase - A firebase app
+ * @returns {Adapter} Adapter
+ */
+export const FirebaseAdapter = (firebase: FirebaseApp): Adapter => {
+  const auth = getAuth(firebase);
+  const db = getFirestore(firebase);
+
+  return {
+    checkNonce: async (pubkey: string) => {
+      const docRef = db.doc(`profiles/${pubkey}`);
+      const doc = await docRef.get();
+
+      if (doc.exists) {
+        return doc.data()?.nonce;
+      }
+      return undefined;
+    },
+
+    verifyTTL: async (pubkey: string) => {
```

Comment:
I don't think the firebase adapter should be in charge of doing the date comparison, because that's independent on where the date is stored

I would refactor this so this function only fetches the TTL, and rename it getTTLForPubkey or something